tools/RestfulApiWebService/RestfulApiWebService/Controllers/QueryController.cs

```diff
@@ -0,0 +1,81 @@
+﻿using Microsoft.DocAsCode.Build.Engine;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks;
+using System.Web.Http;
+using System.Web.Http.Description;
+using System.Data.Entity;
+
+using System.Web.Script.Serialization;
+using Microsoft.DocAsCode.Plugins;
+using DB.Models;
+using System.Diagnostics;
+using DB.Dao;
+using Newtonsoft.Json;
+using DB.Utility;
+using System.IO;
+
+namespace RestfulApiService.Controllers
+{
+    [RoutePrefix("uids")]
+    public class QueryController : ApiController
+    {
+        private XrefSpecDBContext _db = new XrefSpecDBContext();
+
+        [HttpGet]
+        [Route("{uid}")]
+        public async Task<IHttpActionResult> GetByUid(string uid)
+        {   
+            string hashedUid = MD5Encryption.CalculateMD5Hash(uid);
+            var ut = await _db.XRefSpecObjects.Where(b => b.HashedUid == hashedUid)
+                                .Select(c => c.XRefSpecJson)
+                                .ToListAsync();
+
+            var response = new HttpResponseMessage(); 
+            var ms = new MemoryStream();
+            var sw= new StreamWriter(ms);
+            sw.Write("[");
+            sw.Write(string.Join(",", ut));
+            sw.Write("]");
+            sw.Flush();
+            ms.Position = 0;
+            response.Content = new StreamContent(ms);
+            response.StatusCode = HttpStatusCode.OK;
+            return ResponseMessage(response);
+        }
+           
+        [HttpPost]
+        [Route("")]
+        public async Task<IHttpActionResult> PostByUids([FromBody]string[] uids)
+        {
+            List<XRefSpec> xfs = new List<XRefSpec>();
+            var getUidTasks = new List<Task<XRefSpecObject>>();
+            foreach (string uid in uids)
+            {
+                XrefSpecDBContext tempDB = new XrefSpecDBContext();
+                string hashedUid = MD5Encryption.CalculateMD5Hash(uid);
+                getUidTasks.Add(tempDB.XRefSpecObjects.Where(b => b.HashedUid == hashedUid)
+                               .Select(c => c)
+                               .FirstOrDefaultAsync());
+            }
+
+            var uidList = await Task.WhenAll(getUidTasks);
+            foreach(var temp in uidList)
+            {
+                XRefSpec xf = null;
+                if (temp != null)
+                {
+                    xf = JsonConvert.DeserializeObject<XRefSpec>(temp.XRefSpecJson);
```

Comment:
right, but the post method has go out of use

---
In reply to: [131039796](https://github.com/dotnet/docfx/pull/1906#discussion_r131039796) [](ancestors = 131039796)